### ISSUE TITLE
Interchain Accounts genesis state missing when init

### ISSUE THREAD

**Issue by dimiandre** [CosmosContracts/juno#264]
when using 
```junod init --chain-id my-local``` 

interchain accounts state is missing in genesis, the chain can start normally but will result in other errors when trying export the state in the future

error: 

```
% junod export
panic: UnmarshalJSON cannot decode empty bytes

goroutine 1 [running]:
github.com/cosmos/cosmos-sdk/x/params/types.Subspace.Get({{0x65ac198, 0xc000191c70}, 0xc0001ae2d0, {0x6588d70, 0xc0011164b0}, {0x6588dc0, 0xc001116560}, {0xc000eff038, 0x7, 0x14}, ...}, ...)
	github.com/cosmos/cosmos-sdk@v0.45.4/x/params/types/subspace.go:109 +0x307
github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts/host/keeper.Keeper.IsHostEnabled(...)
	github.com/cosmos/ibc-go/v3@v3.0.0/modules/apps/27-interchain-accounts/host/keeper/params.go:13
github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts/host/keeper.Keeper.GetParams({{0x6588d70, 0xc001116530}, {0x65ac198, 0xc000191c70}, {{0x65ac198, 0xc000191c70}, 0xc0001ae2d0, {0x6588d70, 0xc0011164b0}, {0x6588dc0, ...}, ...}, ...}, ...)
	github.com/cosmos/ibc-go/v3@v3.0.0/modules/apps/27-interchain-accounts/host/keeper/params.go:26 +0x148
github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts/host/keeper.ExportGenesis({{0x65a0508, 0xc0001b22f0}, {0x65adaa0, 0xc0012da400}, {{0x0, 0x0}, {0x0, 0x0}, 0x3, {0x0, ...}, ...}, ...}, ...)
	github.com/cosmos/ibc-go/v3@v3.0.0/modules/apps/27-interchain-accounts/host/keeper/genesis.go:38 +0x1b8
github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts.AppModule.ExportGenesis({{}, _, _}, {{0x65a0508, 0xc0001b22f0}, {0x65adaa0, 0xc0012da400}, {{0x0, 0x0}, {0x0, ...}, ...}, ...}, ...)
	github.com/cosmos/ibc-go/v3@v3.0.0/modules/apps/27-interchain-accounts/module.go:181 +0x298
github.com/cosmos/cosmos-sdk/types/module.(*Manager).ExportGenesis(_, {{0x65a0508, 0xc0001b22f0}, {0x65adaa0, 0xc0012da400}, {{0x0, 0x0}, {0x0, 0x0}, 0x3, ...}, ...}, ...)
	github.com/cosmos/cosmos-sdk@v0.45.4/types/module/module.go:341 +0x125
github.com/CosmosContracts/juno/v7/app.(*App).ExportAppStateAndValidators(0xc000ffaa80, 0x0, {0x75e0810, 0x0, 0x0})
	github.com/CosmosContracts/juno/v7/app/export.go:32 +0x20b
github.com/ignite-hq/cli/ignite/pkg/cosmoscmd.appCreator.appExport({{{0x65a4d20, 0xc00019e588}, {0x65affb0, 0xc000191c70}, {0x65a9ca0, 0xc0005b2d00}, 0xc0001ae2d0}, 0x631e328}, {0x65a17a0, 0xc001060a20}, ...)
	github.com/ignite-hq/cli@v0.22.1/ignite/pkg/cosmoscmd/root.go:416 +0x24b
github.com/cosmos/cosmos-sdk/server.ExportCmd.func1(0xc0013aac80?, {0x75e0810?, 0x0?, 0x0?})
	github.com/cosmos/cosmos-sdk@v0.45.4/server/export.go:71 +0x31c
github.com/spf13/cobra.(*Command).execute(0xc0013aac80, {0x75e0810, 0x0, 0x0})
	github.com/spf13/cobra@v1.4.0/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0xc0006c5900)
	github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.4.0/command.go:902
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	github.com/spf13/cobra@v1.4.0/command.go:895
github.com/cosmos/cosmos-sdk/server/cmd.Execute(0x631e328?, {0xc000698168, 0x11})
	github.com/cosmos/cosmos-sdk@v0.45.4/server/cmd/execute.go:36 +0x1eb
main.main()
	github.com/CosmosContracts/juno/v7/cmd/junod/main.go:24 +0x7c
```

missing state: 

```
"interchainaccounts": {
      "controller_genesis_state": {
        "active_channels": [],
        "interchain_accounts": [],
        "params": {
          "controller_enabled": true
        },
        "ports": []
      },
      "host_genesis_state": {
        "active_channels": [],
        "interchain_accounts": [],
        "params": {
          "allow_messages": [
            "/cosmos.bank.v1beta1.MsgSend",
            "/cosmos.staking.v1beta1.MsgDelegate",
            "/cosmos.staking.v1beta1.MsgBeginRedelegate",
            "/cosmos.staking.v1beta1.MsgCreateValidator",
            "/cosmos.staking.v1beta1.MsgEditValidator",
            "/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward",
            "/cosmos.distribution.v1beta1.MsgSetWithdrawAddress",
            "/cosmos.distribution.v1beta1.MsgWithdrawValidatorCommission",
            "/cosmos.distribution.v1beta1.MsgFundCommunityPool",
            "/cosmos.gov.v1beta1.MsgVote",
            "/cosmos.authz.v1beta1.MsgExec",
            "/cosmos.authz.v1beta1.MsgGrant",
            "/cosmos.authz.v1beta1.MsgRevoke",
            "/cosmwasm.wasm.v1.MsgStoreCode",
            "/cosmwasm.wasm.v1.MsgInstantiateContract",
            "/cosmwasm.wasm.v1.MsgExecuteContract"
          ],
          "host_enabled": true
        },
        "port": "icahost"
      }
    },
```
    
    this bug is present since junod v7.0.0

**Comment by faddat** [comment]
this is quality.

I don't know the answer, but will look into it.

**Comment by the-frey** [comment]
we added these in the upgrade handler and wrote them to the store:

```
            "/cosmos.bank.v1beta1.MsgSend",
            "/cosmos.staking.v1beta1.MsgDelegate",
            "/cosmos.staking.v1beta1.MsgBeginRedelegate",
            "/cosmos.staking.v1beta1.MsgCreateValidator",
            "/cosmos.staking.v1beta1.MsgEditValidator",
            "/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward",
            "/cosmos.distribution.v1beta1.MsgSetWithdrawAddress",
            "/cosmos.distribution.v1beta1.MsgWithdrawValidatorCommission",
            "/cosmos.distribution.v1beta1.MsgFundCommunityPool",
            "/cosmos.gov.v1beta1.MsgVote",
            "/cosmos.authz.v1beta1.MsgExec",
            "/cosmos.authz.v1beta1.MsgGrant",
            "/cosmos.authz.v1beta1.MsgRevoke",
            "/cosmwasm.wasm.v1.MsgStoreCode",
            "/cosmwasm.wasm.v1.MsgInstantiateContract",
            "/cosmwasm.wasm.v1.MsgExecuteContract"
    ```
    
    so I would assume those need to be added to the initchain

**Comment by the-frey** [comment]
But, naively would have assumed https://github.com/CosmosContracts/juno/blob/main/app/app.go#L611 would be enough.

Although, I suppose that may be just the very basic initgenesis

**Comment by the-frey** [comment]
I guess the code required is something like:

```go
// we would need to add to consensus versions hash i think
// icatypes.ModuleName = 1

controllerParams := icacontrollertypes.Params{}

hostParams := icahosttypes.Params{
  HostEnabled: true,
  AllowMessages: []string{
    sdk.MsgTypeURL(&banktypes.MsgSend{}),
    sdk.MsgTypeURL(&stakingtypes.MsgDelegate{}),
    sdk.MsgTypeURL(&stakingtypes.MsgBeginRedelegate{}),
    sdk.MsgTypeURL(&stakingtypes.MsgCreateValidator{}),
    sdk.MsgTypeURL(&stakingtypes.MsgEditValidator{}),
    sdk.MsgTypeURL(&distrtypes.MsgWithdrawDelegatorReward{}),
    sdk.MsgTypeURL(&distrtypes.MsgSetWithdrawAddress{}),
    sdk.MsgTypeURL(&distrtypes.MsgWithdrawValidatorCommission{}),
    sdk.MsgTypeURL(&distrtypes.MsgFundCommunityPool{}),
    sdk.MsgTypeURL(&govtypes.MsgVote{}),
    sdk.MsgTypeURL(&authz.MsgExec{}),
    sdk.MsgTypeURL(&authz.MsgGrant{}),
    sdk.MsgTypeURL(&authz.MsgRevoke{}),
    // wasm msgs here
    // note we only support these three for now
    sdk.MsgTypeURL(&wasmtypes.MsgStoreCode{}),
    sdk.MsgTypeURL(&wasmtypes.MsgInstantiateContract{}),
    sdk.MsgTypeURL(&wasmtypes.MsgExecuteContract{}),
  },
}

icamodule, correctTypecast := app.mm.Modules[icatypes.ModuleName].(ica.AppModule)
if !correctTypecast {
  panic("mm.Modules[icatypes.ModuleName] is not of type ica.AppModule")
}
icamodule.InitModule(ctx, controllerParams, hostParams)
```

Based on what we did to init the host.

**Comment by the-frey** [comment]
This is closed by #282 I believe
